MoJoLEG/Presentation/View/ScenarioProps/ScenarioPropsView.swift

```diff
@@ -0,0 +1,390 @@
+    }
+  }
+
+  private var background: some View {
+    Color.gray100
+      .ignoresSafeArea()
+  }
+
+  private var topToolbar: some View {
+    HStack {
+      Button {
+        withAnimation {
+          isSidebarPresented = true
+        }
+      } label: {
+        Image(systemName: "sidebar.left")
+          .foregroundStyle(.primaryYellow)
+      }
+
+      Spacer()
+
+      HStack(spacing: 12) {
+        HStack {
+          Toggle(isOn: $isScenarioPresented.animation(.default)) {}
+            .tint(.primaryYellow)
+            .frame(maxWidth: 72)
+          Text("시나리오 보기")
+            .foregroundStyle(.gray900)
+            .padding(.horizontal, 10)
+        }
+        ShareLink(item: ExcelService.shared.createExcelFile(scenario))
+          .labelStyle(.iconOnly)
+          .foregroundStyle(.primaryYellow)
+      }
+    }
+  }
+
+  private var scenarioTitle: some View {
+    Text(scenario.title)
+      .font(.system(size: 40, weight: .semibold))
+      .foregroundStyle(.gray900)
+  }
+
+  private var layoutPicker: some View {
+    HStack {
+      Button {
+        withAnimation {
+          selectedLayout = .list
+        }
+      } label: {
+        Text("리스트뷰")
+          .foregroundStyle(selectedLayout == .list ? .white : .gray900)
+          .padding(.vertical, 8)
+          .padding(.horizontal, 16)
+          .background {
+            if selectedLayout == .list {
+              Capsule()
+                .fill(.primaryYellow)
+                .matchedGeometryEffect(id: "LayoutBackground", in: namespace)
+                .shadow(color: .black.opacity(0.1), radius: 4, y: 2)
+            }
+          }
+      }
+      Button {
+        withAnimation {
+          selectedLayout = .gallery
+        }
+      } label: {
+        Text("갤러리뷰")
+          .foregroundStyle(selectedLayout == .gallery ? .white : .gray900)
+          .padding(.vertical, 8)
+          .padding(.horizontal, 16)
+          .background {
+            if selectedLayout == .gallery {
+              Capsule()
+                .fill(.primaryYellow)
+                .matchedGeometryEffect(id: "LayoutBackground", in: namespace)
+                .shadow(color: .black.opacity(0.1), radius: 4, y: 2)
+            }
+          }
+      }
+    }
+    .padding(4)
+    .background(.gray200, in: Capsule())
+  }
+
+  private var searchBar: some View {
+    HStack {
+      Image(systemName: "magnifyingglass")
+        .foregroundStyle(.gray400)
+      TextField("검색", text: $searchText)
+      Button {
+        searchText = ""
+      } label: {
+        Image(systemName: "xmark.circle.fill")
+          .foregroundStyle(.gray400)
+      }
+    }
+    .padding(7)
+    .frame(maxWidth: 300)
+    .background(.gray200, in: RoundedRectangle(cornerRadius: 8))
+    .opacity(isSearchBarPresented ? 1.0 : 0.0)
+  }
+
+  private var searchButton: some View {
+    Button {
+      withAnimation {
+        isSearchBarPresented.toggle()
```

Comment:
현재 검색 기능에서 키워드 검색 후 취소버튼을 누르면 searchText가 초기화 되지 않습니다.

모니카 & 그린과 이야기 해보고 굳이 searchText가 searchBar안에 유지될 필요가 없으면 

```suggestion
     if isSearchBarPresented {
        searchText = ""
        isSearchBarPresented = false
      } else {
        isSearchBarPresented = true
      }
```

다음과 같은 형태로 초기화 가능하게 코드 변경하는게 어떨까 제안드립니다!

Reply:
수정했습니다.